apps/dashboard/src/lib/utils/functions/theme.ts

```diff
@@ -1,14 +1,109 @@
-export function setTheme(theme?: string) {
+import { lighten } from 'color2k';
+
+export function setTheme(theme: string) {
+  // this condition is to check if it's a hex code coming from the db or it's a specified theme
+  if (theme && !theme.includes('theme-')) {
+    // this adds the "custom-theme" styles to the head tag
+    injectCustomTheme(theme);
+
+    // this adds the "custom-theme" classname to the head tag 
+    setCustomTheme('theme-custom');
+  } else {
+
+    // if the string passed has "theme" in it, it goes on with the existing logic
+    if (!theme && document.body.className.includes('theme-')) {
+          const regex = /theme-[\w]+/gi;
+          document.body.className = document.body.className.replace(regex, theme || '');
+          return;
+        }
+      
+        // In case theme already exists in dom, don't add
+        if (document.body.className.includes(theme || '')) return;
+      
+        localStorage.setItem('theme', theme || '');
+        document.body.className = document.body.className.concat(' ', theme || '');
+  }
+}
+
+export function setCustomTheme(theme?: string) {
   // In case the default theme is added but another theme exists
   if (!theme && document.body.className.includes('theme-')) {
     const regex = /theme-[\w]+/gi;
     document.body.className = document.body.className.replace(regex, theme || '');
     return;
   }
 
-  // In case theme already exists in dom, don't add
-  if (document.body.className.includes(theme || '')) return;
+  // Remove any class starting with "theme-"
+  const regex = /theme-[\w]+/gi;
+  document.body.className = document.body.className.replace(regex, '');
+
+  // Add the new theme if it doesn't already exist
+  if (!document.body.className.includes(theme || '')) {
+    localStorage.setItem('theme', theme || '');
+    document.body.className = document.body.className.concat(' ', theme || '');
+  }
+}
+
+
+export function injectCustomTheme(hex: string) {
+  const styleId = 'theme-custom';
+  let styleElement = document.getElementById(styleId);
+
+  // generate shades using color2k's lighten function
+  const shades = {
+    50: lighten(hex, 0.9),
+    100: lighten(hex, 0.8),
+    200: lighten(hex, 0.7),
+    300: lighten(hex, 0.6),
+    400: lighten(hex, 0.5),
+    500: lighten(hex, 0.4),
+    600: lighten(hex, 0.3),
+    700: lighten(hex, 0.2),
+    800: lighten(hex, 0.1),
+    900: hex, // the original color
```

Comment:
@tunny17 The original color should actually be 700 because that's what we use everywhere.